src/en/authors-charm-config.md

```diff
@@ -2,25 +2,26 @@ Title: Creating config.yaml and configuring charms
 
 # Charm configuration
 
-The optional `config.yaml` file defines how a service running the charm can be
-configured by its user.
+The optional `config.yaml` file defines how the charm can be configured by the
+user.
 
-If `config.yaml` exists, it must contain a dictionary of `options`, in which
-each possible setting is named by the key and defined by the value. An option
-definition must contain all of the following fields:
+If `config.yaml` is used, it must contain a dictionary of `options`, in which
+every setting is named by the key and defined by the value. An option
+definition must contain the `type` field. Otherwise, an error will be generated
+by the [charm proof tool](tools-charm-tools.html#proof) indicating the option
+is not compliant with Charm Store policy.
 
-  - `type` can be `string`, `int`, `float`, or `boolean`. The default type is
-    `string`.
-  - `description` should contain an explanation of what the user might achieve
-    by altering the setting along with valid values.
-  - `default` should contain a value of the appropriate type. If set as
-    `default:` with no trailing characters, it is treated as null (which is
-    always a valid value in this context).
+The following fields are available:
 
-Any option without these three fields will generate a Warning from the the
-[charm proof tool](tools-charm-tools.html#proof)
-indicating the option is not compliant with charm store policy. This policy
-allows older versions of juju to safely unset values.
+ - `type` : can be `string`, `int`, `float`, or `boolean`. The default type is
```

Comment:
I am not sure if we need to say "The default type is..".
I mean, providing the type is mandatory when defining an option. And when the type is provided, it cannot be anything but {string, int, float, boolean}. Default implies that if you just say 'type' without one of the 4 types, something will be substituted. Whereas I am pretty sure, we will err out. There is no 'default' as such, is there?

Reply:
I think it will be helpful to say here 'Type must be provided" or something to this effect.
And for 'description' and 'default' that they are optional.

Reply:
Agreed.